### ISSUE TITLE
Can't open with Gimp

### ISSUE THREAD

**Issue by wd5gnr** [flameshot-org/flameshot#2708]
### Flameshot Version

12.0rc1 (installed from Ubuntu 20.x deb).

### Installation Type

Using the ready-made package from Github Releases

### Operating System type and version

KDE Neon

### Description

Installed from .deb. Gimp is a flatpak but there is a link to the var/export for the flatpak that will start gimp from the shell (as gimp).

When I try to open with, the Open With dialog shows up and has two Gimp icons, but neither work. Okular, Gwenview, etc. are also there and they work.

### Steps to reproduce

1. Take a screenshot
2. Press Open With button
3. Observe 2 Gimp icons
4. Press either one
5. Gimp does not open

### System Information

Up to date Neon. Gimp as a flatpak with a link in ~/bin that will open the flatpak named gimp.

X11 display.

**Comment by borgmanJeremy** [comment]
Are Okular or Gwenview a flatpak?

**Comment by wd5gnr** [comment]
No. However, the menu shows two copies of Gimp, it just won't run them. In addition there is a script on my path: 

[alw:~] $ cat /home/alw/bin/gimp
───────┬──────────────────────────────────────────────────────────────────────
       │ File: /home/alw/bin/gimp
       │ Size: 85 B
   1   │ #!/bin/sh
   2   │ exec /usr/bin/flatpak run --branch=stable --arch=x86_64 org.gimp.GIMP "$@"

**Comment by borgmanJeremy** [comment]
Can you add a debug statement to your script to print the arguments it is getting and confirm the script is executing at all?

Under the hood flameshot is using QProcess to call your script and pass an argument to a temp file: https://github.com/flameshot-org/flameshot/blob/1f3ca56b977832cbf403a336c51d56a3da55cd8c/src/tools/launcher/applauncherwidget.cpp#L113

**Comment by wd5gnr** [comment]
Yes I have done that. It isn't launched at all. I tried the AppImage version -- same result. I also moved ~/bin/gimp (the script that runs flatpak) to /usr/local/bin and no change. The dialog shows two Gimps (no telling where those are coming from) and neither of them appear to launch anything.

I do have two icons in the app menu and changing the title of one of them did change the title, but zero response. Both icons are attached to something like: flatpak run ....

I deleted one of them and flameshot still shows two, however, it maybe a restart will take away the 2nd one. But nothing even appears to happen when you select one of those. 

I did a little more work. I replace flatpak with a simple script that logs its arguments and then runs the real executable. It looks like the code is grabbing the icon and assuming the program text is a single word (like gimp). Then it puts the tmp file name in. The problem is, it needs to replace properly (%f) and if there is no filename, append it. 

So what's happening is where it should be running **flatpak run org.gimp.GIMP tmp-file-name** you are really running **flatpak tmp-file-name**.

The code you pointed out actually shows this. It runs the first token and then just assumes there are no other arguments other than the file.

I can "fix" this by pointing my desktop files at my script instead of flatpak, but that's a workaround. Would be nicer if the argument list was constructed. I may fix this if I can get a build environment set up for it later in the week.
